### PR TITLE
Addes assert messages to bitmanip

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

File: std/bitmanip.d [hunk 1]
```diff
@@ -1028,8 +1028,10 @@ public:
     this(void[] v, size_t numbits) @nogc nothrow pure
     in
     {
-        assert(numbits <= v.length * 8);
-        assert(v.length % size_t.sizeof == 0);
+        assert(numbits <= v.length * 8,
+            "numbits must be less than or equal to v.length * 8");
```
Comment (atilaneves) [inline std/bitmanip.d]: indentation

File: std/bitmanip.d [hunk 2]
```diff
@@ -1028,8 +1028,10 @@ public:
...
+        assert(v.length % size_t.sizeof == 0,
+            "v.length must be a multiple of the size of size_t");
```
Comment (atilaneves) [inline std/bitmanip.d]: indentation

File: std/bitmanip.d [hunk 3]
```diff
@@ -1233,8 +1235,8 @@ public:
     void opSliceAssign(bool val, size_t start, size_t end)
     in
     {
-        assert(start <= end);
-        assert(end <= length);
+        assert(start <= end, "start must not be greater than end");
+        assert(end <= length, "end must be less or equal to the length");
```
Comment (atilaneves) [inline std/bitmanip.d]: Why the different wording for essentially the same assertion?

Comment (burner) [reply std/bitmanip.d]: fixed

File: std/bitmanip.d (hunk 1 above)
Comment (burner) [reply std/bitmanip.d]: fixed

File: std/bitmanip.d (hunk 3 above)
Comment (burner) [reply std/bitmanip.d]: fixed

**Comment by burner** [comment]
@n8sh thanks

**Pull request merged**